### DISCUSSION TITLE
How to output creep strain rate in the class "PowerLawCreepStressUpdate"

### DISCUSSION THREAD

**Discussion by xchengood** [idaholab/moose#20575] [Q&A Modules: Solid mechanics] [answered]
Hi Moose experts or users,

Could anyone share advice about how to output creep strain rate in **PowerLawCreepStressUpdate** with me? The power-law creep model just describes the secondary steady-state case and it cannot reflect whole creep behavior usually including three stages. Thank you.

**Comment by recuero** [comment, marked as answer]
The creep strain rate in `PowerLawCreepStressUpdate` is an internal local variable, so it can't be outputted as of now. You can output `creep_strain` instead, from which you could compute its rate. 

Yeah, this is a fairly basic model. Blackbear (https://github.com/idaholab/blackbear), a MOOSE app, couples a library with more advanced creep models, in case you are interested.

**Reply by xchengood** [reply]
Hi @recuero. Thank you for your advice. Could you tell me more about if there is an easy way to accurately compute creep strain rate from **creep_strain**? I saw there are many equations reported in the literature, which are not convenient to solve the rate from outputted strain.

**Reply by recuero** [reply]
Probably the most convenient is to compute it from the stress by making use of the power law equation and the input parameters. This class implements the typical A * stress^{n} Norton model with some extra parameters.

**Reply by xchengood** [reply]
I see. Appreciate your help.
